vnext/local-cli/config/dependencyConfig.js

```diff
@@ -1,102 +1,117 @@
-const fs = require('fs');
+/**
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ * @format
+ */
+// @ts-check
+
 const path = require('path');
-const glob = require('glob');
-const xmldoc = require('xmldoc');
 
-function dependencyConfigWindows(folder, userConfig = {}) {
-  const sourceDir = userConfig.sourceDir || findWindowsAppFolder(folder);
+const configUtils = require('./configUtils.js');
+
+/*
+
+Schema for dependencies:
+
+{
+  folder: '', // absolute path to the module root, ex: c:\path\to\node_modules\module-name
+  sourceDir: '', // relative path to the windows implementation folder, ex: windows
+  solutionFile: '', // relative path to the solution, ex: ModuleName.sln
+  projects: [ // array of projects to be built that include modules
+    {
+      projectFile: '', // relative path to the project, ex: ProjectName\ProjectName.vcxproj
+      projectName: '', // name of the project
+      projectLang: '', // language of the project, cpp or cs
+      projectGuid: '', // project identifier
+      cppHeaders: [], // array of cpp header include lines, ie: 'winrt/ModuleName.h', to be transformed into '#include <winrt/ModuleName.h>'
+      cppPackageProviders: [], // array of fully qualified cpp IReactPackageProviders, ie: 'ModuleName::ReactPackageProvider'
+      csNamespaces: [], // array of cs namespaces, ie: 'ModuleName', to be transformed into 'using ModuleName;'
+      csPackageProviders: [], // array of fully qualified cs IReactPackageProviders, ie: 'ModuleName.ReactPackageProvider'
+    },
+  ],
+  additionalProjects: [ // array of (dependency) projects that must be built, but do not contain modules
```

Comment:
Can we simplify this and just allow projects to not have any package providers, then you just need the one list of projects?

Reply:
We need a way of differentiating:
* These projects need to be both in the solution and directly referenced by the app project (then we loop through the package providers, if any, to be added to the project)
* These projects need to be in the solution just so the build works. I could refactor those into a boolean

Reply:
Fixed.